addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -79,8 +79,8 @@ options.registry.TableOfContent = options.Class.extend({
      */
     _generateNav: function (ev) {
         this.options.wysiwyg && this.options.wysiwyg.odooEditor.unbreakableStepUnactive();
-        const headingsEls = this.$target.find(this.targetedElements).toArray()
-            .filter(el => !el.closest('.o_snippet_desktop_invisible'));
+        const targetedEls = this.$target.find(this.targetedElements).toArray();
```

Comment:
Take the opportunity to get rid of JQuery. You can do the same with vanilla JS :+1: